### DISCUSSION TITLE
Updating SALEOR and Production deployment.

### DISCUSSION THREAD

**Discussion by luluhoc** [saleor/saleor#9423] [General]
I'm really interested to use saleor for my next projects but the documentation lacks many things, especially production deployment. Are there any news on how to update saleor in 2022 or still nothing?

**Comment by karolkielecki** [comment]
Hi @luluhoc, could you please be more specific on what's missing? We always provide a guide on how to update Saleor e.g., from 2.11 to 3.0. https://docs.saleor.io/docs/3.x/developer/upgrade-guide/2-11-to-3-0

**Reply by luluhoc** [reply]
> Hi @luluhoc, could you please be more specific on what's missing? We always provide a guide on how to update Saleor e.g., from 2.11 to 3.0. https://docs.saleor.io/docs/3.x/developer/upgrade-guide/2-11-to-3-0

I mean for example on how to update from 3.0.1 to 3.1?

And the biggest problem for me is how to deploy to production.

**Reply by deltacodepl** [reply]
Hi @luluhoc, where do you want to deploy saleor exactly?

**Reply by lack-of-gravitas** [reply]
Some BASIC instructions on how to deploy would be nice including configuration of ENV vars which at the moment is imo atrocious

**Reply by tiendq** [reply]
Personally I think difficulties are in Django, nginx, WSGI configuration but not saleor itself. Saleor configuration document (environment variables) is good enough and if you use correct values then `python3 manage.py run...` will work.

**Comment by Caryntjen** [comment]
Yes @karolkielecki, where could we find some nice documention to deploy on production with Docker? 
There is documentation about the properties: https://docs.saleor.io/docs/3.x/developer/running-saleor/configuration but for example: CACHE_URL or REDIS_URL. default is local, I guess for production you need to install Redis manually and link it correctly? Examples for this for prod?

**Reply by karolkielecki** [reply]
Hi @Caryntjen, there is an old Spectrum thread regarding [deploying Saleor API with Docker](https://spectrum.chat/saleor/development/deploying-saleor-api-with-docker-and-ssl~74a2f215-22c1-4aed-8273-f6286817f6a6). A community member wrote it a while ago, but it might be helpful--please let me know if it helps. 

If you run Saleor locally, Redis is used as the broker for Celery and as Cache. A celery broker is required for production purposes, but it doesn't have to be Redis. It can be Rabbit or AWS SQA.

We want to improve the documentation and provide more tutorials. We currently identify what's missing and what might be helpful for the community.

**Reply by lack-of-gravitas** [reply]
@karolkielecki ideally this would exist in the official / latest documentation so we in 2022 **still** dont have to refer to a thread created 2 years ago....

**Reply by timuric** [reply]
@lack-of-gravitas docs contributions are very welcome!

**Comment by tiendq** [comment]
> I'm really interested to use saleor for my next projects but the documentation lacks many things, especially production deployment. Are there any news on how to update saleor in 2022 or still nothing?

Yes, documentation lacks many things but you can still deploy production. It surely takes lots of time to find and read what you need. You should start with looking into `saleor-platform` and ended with WSGI document on Django and could make it running well on an Unbuntu instance.
